translator/source_code/icone_translator.py

```diff
@@ -353,6 +364,34 @@ def parse_incident(incident):
 
     return feature
 
+#function to validate the incident
+def validate_incident(incident):
+    
+    if not incident or (type(incident) != dict and type(incident) != OrderedDict):
+        logging.warning('incident is empty or has invalid type')
+        return False
+    try:
+        coords = parse_polyline(incident['location']['polyline'])
+        incident['starttime']
+        incident['description']
+        incident['creationtime']
+        incident['updatetime']
+        road_name = incident['location']['street']
+        direction = parse_direction_from_street_name(road_name)
+        if not direction:
+            direction=get_road_direction(coords)
+            if not direction:
+                raise RuntimeError('unable to parse direction from street name or polyline')
```

Comment:
i changed it and rather than raise an exception,I logged a warning, and return False.

_**_if not direction:
                logging.warning(f'Invalid incident with id = {incident.get("@id")}.unable to parse direction from street name or polyline')
                return False_**_